genie-web/src/main/java/com/netflix/genie/web/controllers/JobSearchAspect.java

```diff
@@ -0,0 +1,122 @@
+package com.netflix.genie.web.controllers;
+
+import com.netflix.genie.core.util.MetricsConstants;
+import com.netflix.genie.core.util.MetricsUtils;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Aspect
+@Component
+@Slf4j
+public class JobSearchAspect {
```

Comment:
Class name seems wrong.

Reply:
Yeah started as POC, definitely need to rename and relocate